[runSofa] Make names of scene object parameters and links selectable

sergeiNikolaev opened the pull request:
This fix allows to select and copy parameters names in Sofa GUI interface, which is very useful when setting default values for elements in scene files. Instead of setting parameters names as titles of widgets I created QLabel objects and set them as the first row of the GridLayout manager.

I know that current sofa GUI will be replaced with Qt quick GUI soon, but I guess this modification will still be useful for those who use this version of GUI. I also would to make this change for Qt quick GUI, but all parameters names in QML are covered with "Mouse Area" objects, which don't allow to handle mouse events for text objects under them.

______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

Commit:
make names of parameters and names of links selectable

damienmarchal commented:
Hi @kriptian88,

Thank you very much for your contribution and welcome in the sofa development community. 
I will test it asap.

guparan commented:
You forgot to [ci-build] Damien ;-)

guparan commented:
Just tested it:

Before |  After 1 | After 2 
:-------:|:---------:|:--------:|
![before](https://user-images.githubusercontent.com/1651667/29559563-ba6ef7ee-872f-11e7-901d-d584baf0d8be.png) | ![after 1](https://user-images.githubusercontent.com/1651667/29559566-bd0aeb20-872f-11e7-8859-9455807cd76b.png) | ![after 2](https://user-images.githubusercontent.com/1651667/29615570-b4570754-880e-11e7-84a2-4983a2cf0726.png)

damienmarchal commented:
The feature is useful but the visual result is problematic. Isn't it possible to implement similar behavior without touching the rendering ?

Commit:
remove margins around QLabels that present names

sergeiNikolaev commented:
Hello.
I have removed the margins around QLabel that present parameters names to make the size of components smaller.
Anyway, answering to your questions I could mark these issues.
1. Unfortunately, I am not an expert in Qt. But I guess some rather tricky manipulations with layouts should be done to make the rendering absolutely the same.
2. On different systems the rendering is different. I have a notebook with DebianStretch and desktop with Ubuntu 16.04 and I can say that Sofa GUI differs. The changes I've made, unfortunately, affect GUI differently. So, I guess it depends also on how QtWidgets are rendered on different platforms.

Thus, if rendering structure is critical, may be it is better not to touch it at all.


P.S. I have changed my nickname to make it more reasonable.

guparan commented:
Thanks @sergeiNikolaev, I just updated my first comment with a new screenshot 😉

damienmarchal commented:
Hi @sergeiNikolaev 

Respecting the existing gui rendering is not crucial in itself..but we should try to respect standard guidelines for GUIs. On that aspect it is more common to have QLabel & checkbox/textfield on the same line otherwise this look a bit weird and consume a lot of screen space for nothing. 

EDIT: But if this is too much work I think we can live with that as runSofa should die :)

sergeiNikolaev commented:
Hello @damienmarchal.
Ok, I've got you. No, this is not a lot of work.
I wanted to say that I do not know how to replace Widget title with QLabel and remain rendering system the same. In any other case, I guess, there should be a discussion about positions of these QObjects.

As for me, these QLabels present the names of the parameters and they had been placed instead of titles. In this case it is more natural to see the title a bit upper and then the value of parameters. But ok, I can put them on the same line.

damienmarchal commented:
Well, runSofa was not really following guidelines on that :)

sergeiNikolaev commented:
Hello. 
Sorry for not responding quickly. But after I put QLabel in the same line there are still some problems with elements and grey color areas alignment. I need more time to fix that.
![guipic](https://user-images.githubusercontent.com/28653498/29870965-4bf4bd4c-8d8a-11e7-816e-aacb232ab0f0.jpg)

Commit:
add all items to the single line

sergeiNikolaev commented:
I've talked with Bruno Marques, and he said that the version of Sofa GUI without these changes also has problems with color areas alignment. Taking this into account I guess it's better to create a separate issue about this alignment problem. Thus, this branch is ready for another review.

guparan commented:
[ci-build]

guparan commented:
SOFA-dev meeting report: @damienmarchal will merge this PR with #373 since their contents are related. Doing so, he will try to get rid of this box problem simply by removing all boxes (separating Datas with lines). To be continued...

guparan commented:
Postponed because of lack of time. Do not hesitate to re-open.
